File: elixir/darts/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  defp distance_from_origin({x, y}) do
    :math.sqrt(x * x + y * y)
  end

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    distance_from_center = distance_from_origin({x, y})
    outer_radius = 10
    middle_radius = 5
    inner_radius = 1

    cond do
      distance_from_center <= inner_radius -> 10
      distance_from_center <= middle_radius -> 5
      distance_from_center <= outer_radius -> 1
      distance_from_center > outer_radius -> 0
    end
  end
end

File: go/darts/darts.go
package darts

import "math"

var (
	OUTER_CIRCLE_RADIUS  = 10.0
	MIDDLE_CIRCLE_RADIUS = 5.0
	INNER_CIRCLE_RADIUS  = 1.0
)

func Score(x, y float64) int {
	throwDistanceFromCenter := DistanceFromCenter(x, y)

	if throwDistanceFromCenter > OUTER_CIRCLE_RADIUS {
		return 0
	}
	if throwDistanceFromCenter > MIDDLE_CIRCLE_RADIUS {
		return 1
	}
	if throwDistanceFromCenter > INNER_CIRCLE_RADIUS {
		return 5
	}

	return 10
}

func DistanceFromCenter(x, y float64) float64 {
	return math.Sqrt(x*x + y*y)
}
